virttest/qemu_vm.py

```diff
@@ -155,6 +155,7 @@ def __init__(self, name, params, root_dir, address_cache, state=None):
         self.start_monotonic_time = 0.0
         self.last_boot_index = 0
         self.last_driver_index = 0
+        self.serial_free_port = ''
```

Comment:
After a further look of this PR, I think we need a better implementation that the new tcp protocol support is well based on the original design.